src/js/models/Polygon.js

```diff
@@ -0,0 +1,52 @@
+import { PointInPolygon } from '../math/points';
+
+export default class Polygon {
+    constructor(id, points) {
+        this.points = points;
+        this.id = id;
+        this.intersections = [];
+        this._isIntersected = false;
+    }
+
+    draw(ctx) {
+        if (this.intersections.length !== 0) {
+            this._isIntersected = true;
+        }
+
+        ctx.beginPath();
+        ctx.moveTo(this.points[this.points.length - 1].x, this.points[this.points.length - 1].y);
+        this.points.forEach(n => ctx.lineTo(n.x, n.y));
+        ctx.fillStyle = this.getFillStroke();
+        ctx.fill();
+        ctx.stroke();
+
+    }
+
+    getFillStroke() {
+        if (this._isIntersected) {
+            return '#FF0000';
+        }
+
+        return "#FFFFFF";
+    }
+
+    mouseIn(mouse) {
+        return PointInPolygon(mouse, ...this.points);
+    }
+
+    addFigure(id) {
```

Comment:
`addFigure` - a bad name for the method, because it doesn't explain what the function is doing. You can use for example `addIntersection`